### ISSUE TITLE
I'm unable to configure the cursorline to be slightly transparent.

### ISSUE THREAD

**Issue by GiuseppeMP** [kovidgoyal/kitty#7646]
I was unable to make the Cursor Line slightly transparent instead of fully transparent. As you can see in the image, it appears as a solid dark gray bar.

**To Reproduce**
1. add background_image
2. set background_tint 0.75
3. set background #24283b
4. check Cursorline color

**Actual**
![image](https://github.com/user-attachments/assets/0daf59bd-d80f-4023-8f2e-0e8c5185c4d7)

**Example of what I'm trying to achieve:**
![image](https://github.com/user-attachments/assets/e9c60774-e379-40a1-85f3-9eb4f616be61)

**Environment details**
```
* background_tint 0.75
* background_image
* background #24283b
```

Thank you, folks, for your help in advance!

**Comment by kovidgoyal** [comment]
You cannot. Read the help of the background_opacity option. It works by
making a *single* color semi-transparent.

**Comment by Pouiiro** [comment]
@kovidgoyal Hi sorry for tapping into this again, i saw the fix implemented but really im not sure how to use it for neovim to have the cursor semi transparent using this new configuration! any help would be appreciated!

**Comment by kovidgoyal** [comment]
set the second transparent background color in kitty.conf to the be the value of the
cursor line color you use in nvim.

**Comment by GiuseppeMP** [comment]
@kovidgoyal Ty so for implementing this!

It works fine, but only if I am not using `background_image`. Is this expected?

**Comment by Pouiiro** [comment]
@GiuseppeMP im not using background_image, using full transparency and works for me, cursorLine bg in catppuccin nvim set to same color, 

<img width="1708" alt="Screenshot 2024-08-19 at 7 15 16 PM" src="https://github.com/user-attachments/assets/fca58d93-5754-47c1-a655-3ffb84ec8d9b">

**Comment by GiuseppeMP** [comment]
@kovidgoyal 

Without background image it's working, note that the background image is from desktop:
![image](https://github.com/user-attachments/assets/d2fda51c-d930-44bb-822d-2e667d83f401)

When using a background image in kitty.conf, the `cursorline` becomes fully transparent and, due to the opacity, it blends with the desktop image.
![image](https://github.com/user-attachments/assets/0d4da6c2-2edc-4e5e-b0bd-8d910cf5d5f8)

The `kitty.conf` is in the screenshots.

**Comment by kovidgoyal** [comment]
OK I will take a look at it when I have a moment.

**Comment by dlvhdr** [comment]
I love the new option for `second_transparent_bg` but I would actually like to have more colors be transparent.
My use case is to have my neovim's cursor line semi transparent, as well as git added/removed hunks.
Could this support an array of colors?

See the screenshot for my usecase:
![image](https://github.com/user-attachments/assets/4950da0f-226c-4397-bced-746006420518)

**Comment by dlvhdr** [comment]
I'm talking about 4 colors in total:
- my terminal's background
- red/green for git diff output
- 1 for my neovim cursorline.

**Comment by kovidgoyal** [comment]
Hmm, not too keen as I can kind of see this ballooning endlessly.
Everyone will want to add just another color for this or that use case.

**Comment by dlvhdr** [comment]
Understood. Do any terminals support rgba colors? I know many that support true color, but none that fully support opacity

**Comment by kovidgoyal** [comment]
Not that I know of.

**Comment by kovidgoyal** [comment]
I have decided to remove the second_transparent_bg option and replace it with transparent_background_colors which allows specifying upto seven colors each with its own opacity. For example:

```
transparent_background_colors #ff00aa@0.5 red@0.1 green@0.2
```

This also works with a background image.

This is a breaking change in that the old option is no longer functional, but since it was released only a few weeks ago I decided its not too bad of a break. If you were using second_transparent_bg you will need to update kitty.conf after the next release or if you use kitty nightly.
